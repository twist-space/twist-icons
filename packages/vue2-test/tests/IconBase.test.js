import { mount } from '@vue/test-utils'
import { describe, test, expect } from 'vitest'
import { IconProvider } from '../../_vue2-icons'
import { AiAudioFilled, AiAudioTwotone } from '../../_vue2-icons/ai'

describe('@twist-space/vue2-icons Icon test', () => {
  test('Icon component direct render', () => {
    const Icon = mount(AiAudioFilled)
    expect(Icon.html()).toMatchSnapshot()
  })

  test('Two tone Icon render', () => {
    const TwotoneIcon = mount(AiAudioTwotone)
    expect(TwotoneIcon.html()).toMatchSnapshot()
    // Two tone icon path must than one path
    expect(TwotoneIcon.findAll('path')).not.toHaveLength(1)
  })

  test('Icon component props <size, color, title>', () => {
    const Icon = mount(AiAudioFilled, {
      propsData: {
        size: 20,
        color: 'red',
        title: 'twist-vue2-icon'
      }
    })

    expect(Icon.html()).toMatchSnapshot()
    expect(Icon.html()).toContain('style="color: red;"')
    expect(Icon.html()).toContain('width="20" height="20"')
    expect(Icon.html()).toContain('<title>twist-vue2-icon</title>')
  })

  test('Icon component props <spin>', () => {
    const Icon = mount(AiAudioFilled, {
      propsData: {
        spin: true
      }
    })

    expect(Icon.html()).toMatchSnapshot()
    expect(Icon.classes('twist-icon-loading')).toBe(true)
  })

  test('Icon component props <rotate>', () => {
    const Icon = mount(AiAudioFilled, {
      propsData: {
        rotate: 60
      }
    })

    expect(Icon.html()).toMatchSnapshot()
    expect(Icon.html()).toContain('style="transform: rotate(60deg);"')
  })
})

describe('@twist-space/vue2-icons IconProvider test', () => {
  test('IconProvider direct render', () => {
    const IconProviderWrapper = mount(IconProvider)
    expect(IconProviderWrapper.html()).toMatchSnapshot()
  })
})

describe('IconProvider and Icon component render', () => {
  test('IconProvider nest a Icon component', () => {
    // IconProvider component contains a Icon component,
    // Icon component has a higher priority than IconProvider props
    const Icon = {
      render(h) {
        return h(AiAudioFilled, {
          props: {
            size: 26,
            color: 'green'
          }
        })
      }
    }
    const IconProviderWrapper = mount(IconProvider, {
      propsData: {
        size: 60
      },
      slots: {
        default: Icon
      }
    })

    expect(IconProviderWrapper.html()).toMatchSnapshot()
    expect(IconProviderWrapper.html()).toContain('width="26" height="26"')
    expect(IconProviderWrapper.html()).toContain('color: green;')
  })
})
