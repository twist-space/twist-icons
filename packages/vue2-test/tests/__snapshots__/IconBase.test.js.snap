// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@twistify/vue2-icons Icon test > Icon component direct render 1`] = `
"<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024" width="1em" height="1em" xmlns="http://www.w3.org/2000/svg">
  <path fill="currentColor" d="M512 624c93.9 0 170-75.2 170-168V232c0-92.8-76.1-168-170-168s-170 75.2-170 168v224c0 92.8 76.1 168 170 168zm330-170c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8c0 140.3-113.7 254-254 254S258 594.3 258 454c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8c0 168.7 126.6 307.9 290 327.6V884H326.7c-13.7 0-24.7 14.3-24.7 32v36c0 4.4 2.8 8 6.2 8h407.6c3.4 0 6.2-3.6 6.2-8v-36c0-17.7-11-32-24.7-32H548V782.1c165.3-18 294-158 294-328.1z"></path>
</svg>"
`;

exports[`@twistify/vue2-icons Icon test > Icon component props <rotate> 1`] = `
"<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024" width="1em" height="1em" xmlns="http://www.w3.org/2000/svg" style="transform: rotate(60deg);">
  <path fill="currentColor" d="M512 624c93.9 0 170-75.2 170-168V232c0-92.8-76.1-168-170-168s-170 75.2-170 168v224c0 92.8 76.1 168 170 168zm330-170c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8c0 140.3-113.7 254-254 254S258 594.3 258 454c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8c0 168.7 126.6 307.9 290 327.6V884H326.7c-13.7 0-24.7 14.3-24.7 32v36c0 4.4 2.8 8 6.2 8h407.6c3.4 0 6.2-3.6 6.2-8v-36c0-17.7-11-32-24.7-32H548V782.1c165.3-18 294-158 294-328.1z"></path>
</svg>"
`;

exports[`@twistify/vue2-icons Icon test > Icon component props <size, color, title> 1`] = `
"<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024" width="20" height="20" xmlns="http://www.w3.org/2000/svg" style="color: red;">
  <title>twist-vue2-icon</title>
  <path fill="currentColor" d="M512 624c93.9 0 170-75.2 170-168V232c0-92.8-76.1-168-170-168s-170 75.2-170 168v224c0 92.8 76.1 168 170 168zm330-170c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8c0 140.3-113.7 254-254 254S258 594.3 258 454c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8c0 168.7 126.6 307.9 290 327.6V884H326.7c-13.7 0-24.7 14.3-24.7 32v36c0 4.4 2.8 8 6.2 8h407.6c3.4 0 6.2-3.6 6.2-8v-36c0-17.7-11-32-24.7-32H548V782.1c165.3-18 294-158 294-328.1z"></path>
</svg>"
`;

exports[`@twistify/vue2-icons Icon test > Icon component props <spin> 1`] = `
"<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024" width="1em" height="1em" xmlns="http://www.w3.org/2000/svg" class="twist-icon-loading">
  <path fill="currentColor" d="M512 624c93.9 0 170-75.2 170-168V232c0-92.8-76.1-168-170-168s-170 75.2-170 168v224c0 92.8 76.1 168 170 168zm330-170c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8c0 140.3-113.7 254-254 254S258 594.3 258 454c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8c0 168.7 126.6 307.9 290 327.6V884H326.7c-13.7 0-24.7 14.3-24.7 32v36c0 4.4 2.8 8 6.2 8h407.6c3.4 0 6.2-3.6 6.2-8v-36c0-17.7-11-32-24.7-32H548V782.1c165.3-18 294-158 294-328.1z"></path>
</svg>"
`;

exports[`@twistify/vue2-icons Icon test > Two tone Icon render 1`] = `
"<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024" width="1em" height="1em" xmlns="http://www.w3.org/2000/svg">
  <path fill="currentColor" fill-opacity=".15" d="M512 552c54.3 0 98-43.2 98-96V232c0-52.8-43.7-96-98-96s-98 43.2-98 96v224c0 52.8 43.7 96 98 96z"></path>
  <path fill="currentColor" d="M842 454c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8c0 140.3-113.7 254-254 254S258 594.3 258 454c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8c0 168.7 126.6 307.9 290 327.6V884H326.7c-13.7 0-24.7 14.3-24.7 32v36c0 4.4 2.8 8 6.2 8h407.6c3.4 0 6.2-3.6 6.2-8v-36c0-17.7-11-32-24.7-32H548V782.1c165.3-18 294-158 294-328.1z"></path>
  <path fill="currentColor" d="M512 624c93.9 0 170-75.2 170-168V232c0-92.8-76.1-168-170-168s-170 75.2-170 168v224c0 92.8 76.1 168 170 168zm-98-392c0-52.8 43.7-96 98-96s98 43.2 98 96v224c0 52.8-43.7 96-98 96s-98-43.2-98-96V232z"></path>
</svg>"
`;

exports[`@twistify/vue2-icons IconProvider test > IconProvider direct render 1`] = `"<div></div>"`;

exports[`IconProvider and Icon component render > IconProvider nest a Icon component 1`] = `
"<div><svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024" width="26" height="26" xmlns="http://www.w3.org/2000/svg" style="color: green;">
    <path fill="currentColor" d="M512 624c93.9 0 170-75.2 170-168V232c0-92.8-76.1-168-170-168s-170 75.2-170 168v224c0 92.8 76.1 168 170 168zm330-170c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8c0 140.3-113.7 254-254 254S258 594.3 258 454c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8c0 168.7 126.6 307.9 290 327.6V884H326.7c-13.7 0-24.7 14.3-24.7 32v36c0 4.4 2.8 8 6.2 8h407.6c3.4 0 6.2-3.6 6.2-8v-36c0-17.7-11-32-24.7-32H548V782.1c165.3-18 294-158 294-328.1z"></path>
  </svg></div>"
`;
